client/src/components/Game/Timer.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { withRouter } from 'react-router';
+import { compose } from 'recompose';
+
+class Timer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      minutes: 99,
+      seconds: 99,
+      initialSet: false
+    }
+  }
+
+  async componentDidMount() {
+    const { id } = this.props.match.params;
+    const game = this.props.firebase.game(id);
+
+    game.get()
+    .then(docSnapshot => {
+      if(docSnapshot.exists) {
+        game.onSnapshot((snapshot) => {
+          if(snapshot.data().gameStartTime !== null) {
+            this.setState({
+              gameSize: snapshot.data().players.length,
+              startTime: snapshot.data().gameStartTime.seconds,
+              timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+            })
+            this.setState({
+              timeTurnWillEnd: ((this.state.timePerTurnInSeconds * this.state.gameSize) + this.state.startTime),
+              currentTime: Math.floor(new Date().getTime() / 1000),
+            })
+            this.setState({
+              minutes: Math.floor((this.state.timeTurnWillEnd - this.state.currentTime)/60),
+              seconds: (this.state.timeTurnWillEnd - this.state.currentTime)%60
+            })
+          }
+        })
+      }
+    })
+
+    this.myInterval = setInterval(async () => {
+      const { seconds, minutes } = this.state
+      const gameRef = await this.props.firebase.game(id).get();
+
+      if (seconds > 0) {
+        this.setState(({ seconds }) => ({
+          seconds: seconds - 1
+        }))
+      }
+      if (seconds === 0) {
+        if (minutes === 0) {
+          clearInterval(this.myInterval)
+          game.set({
+            currentPlayerIndex: gameRef.data().currentPlayerIndex + 1,
+          }, { merge: true })
+        } else {
+          this.setState(({ minutes }) => ({
+            minutes: minutes - 1,
```

Comment:
Are there any advantages to manually doing the math each second as opposed to getting the current time with Date()?

Reply:
im using the calculated timeLimit winding down, so i don't think using Date would help me do that

Reply:
timeLeft = timeLimit - date?

Reply:
actually you're right, that's how i am calculating the time in the code above, so really there aren't any differences. that's what initially had and just worked off that, is there an advantage to using date() instead?

Reply:
No need for redundant time arithmetic (we can abstract away converting seconds into minutes), easier to read and see purpose of code, consistent format throughout the code for calculating time, more concise. Also has [existing example](https://codepen.io/SitePoint/pen/NWxKgxN).